### ISSUE TITLE
Stopped working

### ISSUE THREAD

**Issue by hmlendea** [szszoke/sennheiser-gsp670-pulseaudio-profile#5]
Hi, just letting you know this stopped working after updating my Arch Linux system to the lastest packages

Output of `systemctl --user status pulseaudio.service`:
```
● pulseaudio.service - Sound Service
     Loaded: loaded (/usr/lib/systemd/user/pulseaudio.service; disabled; vendor preset: enabled)
     Active: active (running) since Thu 2020-10-15 11:59:03 EEST; 46s ago
TriggeredBy: ● pulseaudio.socket
   Main PID: 882 (pulseaudio)
     CGroup: /user.slice/user-1000.slice/user@1000.service/pulseaudio.service
             ├─882 /usr/bin/pulseaudio --daemonize=no --log-target=journal
             └─925 /usr/lib/pulse/gsettings-helper

oct 15 11:59:03 NuciTop systemd[797]: Starting Sound Service...
oct 15 11:59:03 NuciTop pulseaudio[882]: Failed to find a working profile.
oct 15 11:59:03 NuciTop pulseaudio[882]: Failed to load module "module-alsa-card" (argument: "device_id="1" name="usb-Sennheiser_Sennheiser_GSA_70-00" card_name="alsa_card.usb-Sennheiser_Sennheiser_GSA_70-00" namereg_fail=false tsched=yes fixed_latency_range=no ignore_dB=no deferred_volume=yes use_ucm=yes avoid_resampling=no card_properties="module-udev-detect.discovered=1""): initialization failed.
oct 15 11:59:03 NuciTop pulseaudio[882]: module-rescue-stream is obsolete and should no longer be loaded. Please remove it from your configuration.
oct 15 11:59:03 NuciTop pulseaudio[882]: Module "module-udev-detect" should be loaded once at most. Refusing to load.
oct 15 11:59:03 NuciTop systemd[797]: Started Sound Service.

```

I'm trying to see what's wrong and if I find anything I'll post back here

**Comment by hmlendea** [comment]
Found the problem

The path `/usr/share/pulseaudio/alsa-mixer` needs to be changed `/usr/share/alsa-card-profile/mixer/`

I've updated the PKGBUILD with the new paths and built the packaged, installed it and it works well again

**Comment by hmlendea** [comment]
Here's the PKGBUILD I used: https://github.com/hmlendea/PKGBUILDs/blob/master/pkg/sennheiser-gsp670-pulseaudio-profile/PKGBUILD

I'm not familiar with `.deb` but that one may also need to be updated

**Comment by szszoke** [comment]
Thanks for the help! I will push a new release with the new paths.

**Comment by szszoke** [comment]
I'm not running anything Debian at the moment, but let's see if someone reports issues. I will not change the DEB package right now.

**Comment by szszoke** [comment]
There is a problem though. Manjaro is still using the old paths so I can't just update the paths in `PKGBUILD`.

**Comment by hmlendea** [comment]
Oh ok. In any case, surely it won.t be long until Manjaro will use the
latest packges. Until then it can stay as-is.

sâm., 17 oct. 2020, 14:37 Szabolcs Szőke <notifications@github.com> a scris:

**Comment by pavel-kirienko** [comment]
> Manjaro is still using the old paths so I can't just update the paths in `PKGBUILD`.

As of today, Manjaro is using the new paths so the old package is no longer functional. Applying the fix suggested by @hmlendea fixes the problem.

**Comment by szszoke** [comment]
Thanks for the heads-up!

I will push a new release sometime this week or next week.

**Comment by hmlendea** [comment]
Seems like it broke again even with what initially fixed it.
I ended up installing the files in both locations.

[PKGBUILD example](https://github.com/hmlendea/PKGBUILDs/blob/f2f50cc73969cbb8bbb2c286e1dc255b3420a5db/pkg/sennheiser-gsp670-pulseaudio-profile/PKGBUILD)

It ain't the prettiest solution, but I needed to get this working ASAP as I was unable to enter calls for work. Anyway, seems to work well as far as using it goes.

**Comment by rippin93** [comment]
Hello everyone,

as of this Monday (after the most recent firmware update of the headset and the dongle), the headset's microphone as well as the main output is no longer detected. Running a Debian 10 system with alsa / pulseaudio.

Working for me is `/proc/asound/card1/pcm0p`, not working are `/proc/asound/card1/pcm0c` and `/proc/asound/card1/pcm1p`.

`pacmd list-sources` output (stripped off other input sources):

```
[...]
    index: 6
	name: <alsa_output.usb-Sennheiser_EPOS_GSA_70-00.analog-stereo.monitor>
	driver: <module-alsa-card.c>
	flags: DECIBEL_VOLUME LATENCY DYNAMIC_LATENCY
	state: SUSPENDED
	suspend cause: IDLE
	priority: 1040
	volume: front-left: 65536 / 100% / 0,00 dB,   front-right: 65536 / 100% / 0,00 dB
	        balance 0,00
	base volume: 65536 / 100% / 0,00 dB
	volume steps: 65537
	muted: no
	current latency: 0,00 ms
	max rewind: 0 KiB
	sample spec: s16le 2ch 48000Hz
	channel map: front-left,front-right
	             Stereo
	used by: 0
	linked by: 0
	configured latency: 0,00 ms; range is 0,50 .. 2000,00 ms
	monitor_of: 5
	card: 0 <alsa_card.usb-Sennheiser_EPOS_GSA_70-00>
	module: 6
	properties:
		device.description = "Monitor of EPOS GSA 70 Analog Stereo"
		device.class = "monitor"
		alsa.card = "1"
		alsa.card_name = "EPOS GSA 70"
		alsa.long_card_name = "Sennheiser EPOS GSA 70 at usb-0000:00:14.0-2, full speed"
		alsa.driver_name = "snd_usb_audio"
		device.bus_path = "pci-0000:00:14.0-usb-0:2:1.0"
		sysfs.path = "/devices/pci0000:00/0000:00:14.0/usb1/1-2/1-2:1.0/sound/card1"
		udev.id = "usb-Sennheiser_EPOS_GSA_70-00"
		device.bus = "usb"
		device.vendor.id = "1395"
		device.vendor.name = "Sennheiser Communications"
		device.product.id = "0300"
		device.product.name = "EPOS GSA 70"
		device.serial = "Sennheiser_EPOS_GSA_70"
		device.string = "1"
		module-udev-detect.discovered = "1"
		device.icon_name = "audio-card-usb"
```

If you need any other information, I'll be happy to give them to you.

**Comment by szszoke** [comment]
Nintendo Switch support was recently added to the dongle and it It seems that the USB device ID was also changed.

The current PulseAudio profile is not being used for the updated dongle.
I will try to add the udev rules for the new USB device ID and see if it helps.

**Comment by szszoke** [comment]
TL;DR: If you really need the headset to work, you have to connect it via USB for now, otherwise read on.

Adding a udev rule for the new USB device ID was not enough but it was still required:
```
# /etc/udev/rules.d/91-pulseaudio-sennheiser-gsp670.rules
ATTRS{idVendor}=="1395", ATTRS{idProduct}=="0300", ENV{PULSE_PROFILE_SET}="sennheiser-gsp670-usb-audio.conf"
```

I also noticed that the USB endpoints on the dongle were swapped, so the communication device is now `hw:1,0` and the main device is `hw:0,0`. The `device-strings` have to be updated in `sennheiser-gsp670-usb-audio.conf`:

```
[General]
auto-profiles = yes

[Mapping analog-comm]
description = Communication
device-strings = hw:%f,1,0
channel-map = mono
paths-input = sennheiser-gsp670-input-comm
paths-output = sennheiser-gsp670-output-comm

[Mapping analog-main]
description = Main
device-strings = hw:%f,0,0
channel-map = left,right
paths-output = sennheiser-gsp670-output-main
direction = output

[Profile output:analog-comm+output:analog-main+input:analog-comm]
output-mappings = analog-comm analog-main
input-mappings = analog-comm
priority = 5100
skip-probe = yes
```

The headset is still using the old USB endpoints, so we need two different profiles now in order to support both the GSA-70 dongle and the headset connected directly via USB.

The microphone is still not working. I have to work some more on that.

**Comment by szszoke** [comment]
Something weird is going on with the microphone input. I tried to record a sample via `arecord`. It is failing for the dongle but works for the headset.

```
$ arecord -l
**** List of CAPTURE Hardware Devices ****
card 0: E70 [EPOS GSA 70], device 0: USB Audio [USB Audio]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 1: USB [Scarlett 2i2 USB], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
card 3: Generic [HD-Audio Generic], device 0: ALC1220 Analog [ALC1220 Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 3: Generic [HD-Audio Generic], device 2: ALC1220 Alt Analog [ALC1220 Alt Analog]
  Subdevices: 1/1
  Subdevice #0: subdevice #0
card 4: Camera [Trust USB Camera], device 0: USB Audio [USB Audio]
  Subdevices: 0/1
  Subdevice #0: subdevice #0
```

```
$ arecord -d 5 -f S16_LE -r 16000 --device="hw:0,0" rec.wav
arecord: main:830: audio open error: No such file or directory
```

It looks like the device is found but then it isn't?

**Comment by szszoke** [comment]
In the meantime, if you want to try to do some troubleshooting yourself, all the changes that I made are in the [new-dongle-firmware](https://github.com/szszoke/sennheiser-gsp670-pulseaudio-profile/tree/new-dongle-firmware) branch.

I also [posted](https://forum.manjaro.org/t/sennheiser-gsa-70-arecord-lists-usb-sound-card-but-cannot-record-audio-open-error-no-such-file-or-directory/53511) about this on the Manjaro Forum, hoping that someone can help me in advancing the troubleshooting because as of now I am out of ideas.

**Comment by rippin93** [comment]
Thanks for the effort, I will go ahead and try a bit on my own. For work though, I will take the chance on connecting via cable (at least I can use my good mic instead of the inbuilt one for conferences).

**Comment by szszoke** [comment]
See #18
